ateam_geometry/test/intersection_test.cpp

```diff
@@ -166,3 +166,71 @@ TEST(IntersectionTests, ArcToSegment)
           PointIsNear(ag::Point(-0.707, -0.707)),
           PointIsNear(ag::Point(0.707, 0.707))))));
 }
+
```

Comment:
should this include a test with colinear lines with exactly one intersection point?

E.g.

```
  TEST(IntersectionTests, SegmentToSegmentColinearNoIntersection)
  {
    ag::Segment a{
      ag::Point{0, 0},
      ag::Point{1, 1}
    };
  
    ag::Segment b{
      ag::Point{1, 1},
      ag::Point{2, 2}
    };
  
      EXPECT_THAT(
      ag::intersection(a, b),
      Optional(VariantWith<ag::Point>(PointIsNear(ag::Point{1, 1}))));
  }
```

Reply:
Turns out it should because there was definitely a bug where this test didn't pass. It does now though.